core/regions/src/main/java/software/amazon/awssdk/regions/servicemetadata/EnhancedS3ServiceMetadata.java

```diff
@@ -93,7 +95,8 @@ private boolean useUsEast1RegionalEndpoint(Supplier<ProfileFile> profileFile, Su
             return REGIONAL_SETTING.equalsIgnoreCase(profile);
         }
 
-        return false;
+        return config.advancedOption(ServiceMetadataAdvancedOption.S3_US_EAST_1_REGIONAL_ENDPOINT)
+                     .filter(REGIONAL_SETTING::equalsIgnoreCase).isPresent();
```

Comment:
Usually directly-configured options are higher priority than environment variables and system properties. I know in this case it's actually a default instead of a directly configured option (the customer can't configure this directly, right?), but should we name it accordingly to clarify it's the default, not the actual setting?

Reply:
Yeah, good point. Customers can't configure this if they are using the SDK client. I'll rename it to reflect that it's the default.